### PR TITLE
Feature/242/emailvalidation

### PR DESCRIPTION
**Pull request by andreaMT15**
### Checklist For Submitter

* [x] I have documented all new functionality
* [x] I have screenshotted new UI changes and attached them to this PR
Screen Shot 2019-05-16 at 7 08 07 PM

### CODE REVIEW

**Comment by vhoof** [comment]
For whatever reason, the linter wasn't previously reporting errors. I merged the staging branch into this branch (so you'll want to `git pull` next time you're working on this), which pulled in Henry's linter fix, which means that this PR is now failing CircleCI. It looks to be pretty much entirely stylistic changes, like the use of double quotes and such. Should be fairly straight-forward to fix.

File: frontend/src/components/SignIn/SignIn.js [hunk 1]
```diff
@@ -58,38 +58,43 @@ class SignIn extends Component {
               id="inputSignInEmail"
               className=" sign-in-input fullWidth form-control-lg"
               name="email"
+              pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$"
```
Comment (rmarro) [inline frontend/src/components/SignIn/SignIn.js]: cool, looks ok to me

**Comment by rmarro** [comment]
I think all your code for your other ticket (Feature/247/passwordresetpage) also made it into this branch? There are a ton of changes related to password reset even though I think this feature should just be the one regex that you added

File: frontend/src/components/SignIn/SignIn.js [hunk 2]
```diff
@@ -58,38 +58,43 @@ class SignIn extends Component {
...
-              type={isPasswordVisible ? 'text' : 'password'}
+              type={isPasswordVisible ? "text" : "password"}
               id="inputSignInPassword"
               className="sign-in-input fullWidth form-control-lg"
               name="password"
               required=""
               placeholder="Password"
-              onFocus={event => (event.target.setAttribute('placeholder', ''))}
-              onBlur={event => (event.target.setAttribute('placeholder', 'Password'))}
+              onFocus={event => event.target.setAttribute("placeholder", "")}
```
Comment (SamDesai333) [inline frontend/src/components/SignIn/SignIn.js]: Is it your linter that is making all the single quotes into double quotes? Other than that, everything looks good!

File: frontend/src/components/SignUp/SignUp.js [hunk 3]
```diff
@@ -1,47 +1,47 @@
...
-          errorDiv.setAttribute('class', 'errorMessage sign-up');
-          errorDiv.innerHTML = 'A user with that email already exists! Try signing in.';
-          parentForm.insertAdjacentElement('beforeend', errorDiv);
+      .catch(error => {
+        if (error.message.includes("422")) {
+          const parentForm = document.getElementsByTagName("form")[0];
+          const errorDiv = document.createElement("p");
+          errorDiv.setAttribute("class", "errorMessage sign-up");
+          errorDiv.innerHTML = "A user with that email already exists! Try signing in.";
+          parentForm.insertAdjacentElement("beforeend", errorDiv);
```
Comment (SamDesai333) [inline frontend/src/components/SignUp/SignUp.js]: Catch error looks good to me!

**Pull request closed without merging**